File: data/multilang_comparison/program_go/src/func6.c
#include "program_go.h" 
array_t* func6(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array191 = func7(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array191->id);
      free(params0.data);
      array191->refC--;
      if(array191->refC == 0) {
         free(array191->data);
         free(array191);
         DEBUG_FREE(array191->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array201 = func8(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array201->id);
      free(params0.data);
      array201->refC--;
      if(array201->refC == 0) {
         free(array201->data);
         free(array201);
         DEBUG_FREE(array201->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array206 = func9(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array206->id);
   free(params0.data);
   return array206;
}

File: data/multilang_comparison/program_go/src/func6.go
package main
import "math/rand"
var _ = rand.Intn
func Func6(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array159 *Array = Func7(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array159.RefC--
      if array159.RefC == 0 {
          array159.Data = nil
          array159 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array169 *Array = Func8(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array169.RefC--
      if array169.RefC == 0 {
          array169.Data = nil
          array169 = nil
      }
   }
   var params0 ArrayParam
   params0.Size = 0
   params0.Data = make([]*Array, params0.Size)
   var array174 *Array = Func9(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   return array174;
}

